src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -117,6 +117,20 @@ public void iCanMockPaymentNotificationRequest() throws JsonProcessingException
 
     }
 
+    @And("I can mock payment notification request with missing values")
+    public void iCanMockPaymentNotificationRequestwithMissingValues() throws JsonProcessingException {
+        StringBuilder jsonBuilder = new StringBuilder();
```

Comment:
Do not use JSON as a string approach. Please create a DTO and then convert it to JSON. We did this kind of JSON string in the past and it has become a major technical debt.

Reply:
this is creating a new dto, in a below line the json is getting assigned to the dto